src/LowerTriangularMatrices/lower_triangular_array.jl

```diff
@@ -221,6 +221,14 @@ end
     return getindex(L.data, k, I[3:end]...)
 end
 
+@inline function Base.getindex(L::LowerTriangularArray{T, N}, i::Integer, j::Integer, c::CartesianIndex) where {T, N}
```

Comment:
@maximilian-gelbrecht I've added this method because `L[l, m, k]` is quite a common access pattern arising in

```julia
    for k in eachmatrix(spec)
        for m in 1:mmax
            for l in m:lmax 
                L[l, m, k]
```

to loop over integers `l, m` but any additional dimensions `k` (`eachmatrix` loops over all of those, returning a `CartesianIndex`. I added the type contraints of `i::Integer, j::Integer, c::CartesianIndex` to avoid any dispatch issues, but not sure they're actually needed, e.g. `i, j, c::CartesianIndex` could work equally well.

Reply:
I think `i, j, c::CartesianIndex` might cause some problems with the `getindex` functions that use `Vararg`. I could be mistaking though. 

Anyway, good to a function like this

Reply:
Turns out this method wasn't needed, but we had a `@boundscheck M == N+1` in getindex with varargs which you added to guarantee that someones wants to do matrix indexing with e.g. 2 indices on a LowerTriangularMatrix. But checking this from the number of indices isn't helpful because `L[1, 1, 1, 1, 1, 1]` is the same as `L[1]` regardless the shape of `L` and also you can add many trailing `CartesianIndex()` that would increase the number of indices even though they don't actually count. Just removed that `@boundscheck` to see what happens, I've tested it manually it doesn't seem necessary!